### ISSUE TITLE
[Fabric  & Forge] Inventorio mod

### ISSUE THREAD

**Issue by Laifsyn** [ZsoltMolnarrr/BetterCombat#42]
If you have both [Inventorio](https://www.curseforge.com/minecraft/mc-mods/inventorio) and BetterCombat, when you press F, the item in offhand seems to be deleted. Removing either Inventorio or BetterCombat stops the item from disappearing from offhand - I might submit another issue report at Inventorio too though

recorded testing:https://youtu.be/sZOl_cXWgPU

**Comment by Laifsyn** [comment]
Mods version:
inventorio-1.18-fabric-1.7.0
bettercombat-1.0.10+1.18.2

Fabric loader 0.14.8+

**Comment by Laifsyn** [comment]
```less
08.08 04:47:12 [Server] Server thread/ERROR Failed to handle packet net.minecraft.class_2824@63e3c3ed, suppressing error
08.08 04:47:12 [Server] INFO java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 2
08.08 04:47:12 [Server] INFO at net.bettercombat.logic.PlayerAttackHelper.selectAttack(PlayerAttackHelper.java:63) ~[bettercombat-1.0.10+1.18.2.jar:?]
08.08 04:47:12 [Server] INFO at net.bettercombat.logic.PlayerAttackHelper.getCurrentAttack(PlayerAttackHelper.java:46) ~[bettercombat-1.0.10+1.18.2.jar:?]
08.08 04:47:12 [Server] INFO at net.minecraft.class_1657.localvar$zkm000$disableSweeping(class_1657.java:7851) ~[server-intermediary.jar:?]
08.08 04:47:12 [Server] INFO at net.minecraft.class_1657.method_7324(class_1657.java:1227) ~[server-intermediary.jar:?]
08.08 04:47:12 [Server] INFO at net.minecraft.class_3222.method_7324(class_3222.java:1438) ~[server-intermediary.jar:?]
08.08 04:47:12 [Server] INFO at net.minecraft.class_3244$1.method_34218(class_3244.java:1369) ~[server-intermediary.jar:?]
08.08 04:47:13 [Server] INFO at net.minecraft.class_2824$1.method_34213(class_2824.java:166) ~[server-intermediary.jar:?]
08.08 04:47:13 [Server] INFO at net.minecraft.class_2824.method_34209(class_2824.java:66) ~[server-intermediary.jar:?]
.....
```

I found this on my server logs. idk if it helps. Do note that this isn't from the youtube video, but on the server where I discovered the bug in the first place

**Comment by ZsoltMolnarrr** [comment]
There is a chance, that this is a semantically unresolvable conflict.
Additional investigation needed.

**Comment by ZsoltMolnarrr** [comment]
I am unable to resolve the conflict, without a deep knowledge of how Inventorio works.
Better Combat requires its own mixins to make sure two-handed weapons and dual wielding works. Inventorio seems to fundamentally change the offhand slot.

I am affraid I have to add Invetorio as an explicit incompatibility for Better Combat.

**Comment by ZsoltMolnarrr** [comment]
To make sure people don't lose their items due to the bugs, I will add breaking incompatibility for Inventorio.
I hope we can resolve the issue in the future.

**Comment by ZerbieSlug** [comment]
Currently trying out the [Forge] versions of this mod and Inventorio, seems like this issue still persists/resurfaces as of the most recent 1.19.2 mod versions. (Downloading the Forge Versions of both mods doesn't seem to have the intentional incompatibility crash as the Fabric versions)

Should probably also implement the incompatibility crash for that version as well.
The devs did say they're considering on making Inventorio compatible with Better Combat in a later update, so we'll just have to wait and see...

**Comment by ZsoltMolnarrr** [comment]
Forge mod loader doesn't have breaking incompatibility feature.
I will not develop such thing, since I consider this problem a shortcoming of the mod loader itself.
Using Forge mod loader is not recommended either way.

**Comment by Lizard-Of-Oz** [comment]
Hey @ZsoltMolnarrr, I finally figured out how to make Better Combat compatible with Inventorio thanks to falling_colud96. I pushed the necessary changes to my mod, all you need to do is to change
```js
  "breaks": {
    "inventorio": "*"
  },
```
to
```js
  "breaks": {
    "inventorio": "<1.9.0"
  },
```
in Better Combat's config file.

**Comment by ZsoltMolnarrr** [comment]
Hello @Lizard-Of-Oz !

Thank you very much for the changes. This is a widely requested compatibility fix.
I would like to make sure I only lift breaks flag when the issue is solved for sure.

Could you please elaborate a bit on what scenarios did you test?

**Comment by Lizard-Of-Oz** [comment]
The widely reported bug was items disappearing upon switching hands with a hotkey. This is the issue I fixed - I've intercepted the vanilla hand swap pipeline and replaced it with my own.

Since I don't know Better Combat much, I just tried to swing weapons around and switch between slots and see if it breaks. Better Combat appears to engage upon the second attack when used from the Tool Belt - not ideal, but _fine_.
